File: objective-c/Database/iOS/library/Classes/EditController.m
// **********************************************************************
//
// Copyright (c) 2003-present ZeroC, Inc. All rights reserved.
//
// **********************************************************************

#import <EditController.h>

@implementation EditController

@synthesize textField;
@synthesize initialValue;

-(void)viewDidLoad
{
    textField.placeholder = self.title;
    textField.text = initialValue;
}

-(void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning]; // Releases the view if it doesn't have a superview
    // Release anything that's not essential, such as cached data
}

-(void)startEdit:(NSString*)name value:(NSString*)v
{
    self.title = name;
    self.initialValue = v;
}

@end

File: swift/iOS/library/EditController.swift
//
// Copyright (c) ZeroC, Inc. All rights reserved.
//

import UIKit

class EditController: UIViewController {
    @IBOutlet var textField: UITextField!
    var initialValue: String!

    override func viewDidLoad() {
        super.viewDidLoad()
        textField.placeholder = title
        textField.text = initialValue
    }

    func startEdit(_ name: String, value: String) {
        title = name
        initialValue = value
    }
}
